cli/src/generate/grammar_files.rs

```diff
@@ -436,6 +449,10 @@ fn lookup_package_json_for_path(path: &Path) -> Result<(PathBuf, PackageJSON)> {
         }
         pathbuf.pop(); // package.json
         if !pathbuf.pop() {
+            if path.exists() {
```

Comment:
I disagree with this part, I'd almost say the `tree-sitter` section is necessary for a lot of stuff, especially for the cli. Language configurations can't be looked up without this, and now having written an entry with `file-types` beforehand should alleviate the need for this blob now

Reply:
This is just a check that we are using a `package.json` that already exists in the current directory (if the file does not already exist, we will not create it). Or is it the comment in the following line that's confusing? The comment is supposed to mean "If the package.json is in the current directory (rather than in parent directories), it doesn't need to have a `tree-sitter` section because in that case we will create the section." (if the package.json is in parent dirs, it needs to have a `tree-sitter` section/we will not create the section and will just exit with the error message) -- If that's the case I'll reword the comments.

Reply:
But wasn't the section written to the file *before* this lookup is done though? It *will* exist due to the code you added where the `tree-sitter` section is added to the end of `package.json`, so this shouldn't be needed I think? I might be missing something though